test/rubygems/test_gem_source_github_problem.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+require_relative 'helper'
+require 'rubygems/source'
+
+class TestGemSourceGithubProblem < Gem::TestCase
+  def setup
+    super
+
+    @gem_repo = "https://username:secure_token@rubygems.pkg.github.com/username"
+
+    spec_fetcher
+
+    @source = Gem::Source.new(@gem_repo)
+  end
+
+  def test_load_specs_for_prerelease_0
```

Comment:
When fetching `prerelease` and response contains a gem with version `0` from a Github Rubygems server.